google/auth/transport/_httplib2.py

```diff
@@ -0,0 +1,221 @@
+_STREAM_PROPERTIES = ('read', 'seek', 'tell')
+
+
+class _Response(transport.Response):
+    """httplib2 transport response adapter.
+
+    Args:
+        response (httplib2.Response): The raw httplib2 response.
+        data (bytes): The response body.
+    """
+    def __init__(self, response, data):
+        self._response = response
+        self._data = data
+
+    @property
+    def status(self):
+        return self._response.status
+
+    @property
+    def headers(self):
+        return self._response
+
+    @property
+    def data(self):
+        return self._data
+
+
+class Request(transport.Request):
+    """httplib2 request adapter.
+
+    This class is used internally for making requests using various transports
+    in a consistent way. If you use :class:`AuthorizedHttp` you do not need
+    to construct or use this class directly.
+
+    This class can be useful if you want to manually refresh a
+    :class:`~google.auth.credentials.Credentials` instance::
+
+        import google.auth.transport.httplib2
+        import httplib2
+
+        http = httplib2.Http()
+        request = google.auth.transport.httplib2.Request(http)
+
+        credentials.refresh(request)
+
+    Args:
+        http (httplib2.Http): The underlying http object to use to make
+            requests.
+
+    .. automethod:: __call__
+    """
+    def __init__(self, http):
+        self.http = http
+
+    def __call__(self, url, method='GET', body=None, headers=None,
+                 timeout=None, **kwargs):
+        """Make an HTTP request using httplib2.
+
+        Args:
+            url (str): The URI to be requested.
+            method (str): The HTTP method to use for the request. Defaults
+                to 'GET'.
+            body (bytes): The payload / body in HTTP request.
+            headers (Mapping[str, str]): Request headers.
+            timeout (Optional[int]): The number of seconds to wait for a
+                response from the server. This is ignored by httplib2 and will
+                issue a warning.
+            kwargs: Additional arguments passed throught to the underlying
+                httplib2 :meth:`request` method.
+
+        Returns:
+            google.auth.transport.Response: The HTTP response.
+
+        Raises:
+            google.auth.exceptions.TransportError: If any exception occurred.
+        """
+        if timeout is not None:
+            _LOGGER.warning(
+                'httplib2 transport does not support per-request timeout. '
+                'Set the timeout when constructing the httplib2.Http instance.'
+            )
+
+        try:
+            _LOGGER.debug('Making request: %s %s', method, url)
+            response, data = self.http.request(
+                url, method=method, body=body, headers=headers, **kwargs)
+            return _Response(response, data)
+        except httplib2.HttpLib2Error as exc:
+            raise exceptions.TransportError(exc)
+
+
+def _make_default_http():
```

Comment:
Docstring? (I know it's obvious.) But if you don't do that, you might as well just

```python
_make_default_http = httplib2.Http
```

(though the current impl. guarnatees you won't be able to pass any arguments in)